public/src/actions/campaign.jsx

```diff
@@ -16,6 +16,13 @@ export function setCurrentCampaign(campaignDataObj) {
   };
 }
 
+export function setUserCampaignJoin(response) {
+  return {
+    type: SET_USER_CAMPAIGN_JOIN,
+    payload: !!response
+  };
+}
+
 export function saveNewCampaign(campaignInfo, history) {
   const { name, title, description, script_id, contact_lists_id } = campaignInfo[0];
   const status = campaignInfo[1];
```

Comment:
NIT: I think you could do ` [{ name, title, description, script_id, contact_lists_id }, status ] = campaignInfo `